cognite_toolkit/_cdf_tk/loaders/_resource_loaders/extraction_pipeline_loaders.py

```diff
@@ -293,6 +294,28 @@ def safe_read(self, filepath: Path | str) -> str:
         # The user typically writes the config as an object, so add a | to ensure it is parsed as a string.
         return stringify_value_by_key_in_yaml(safe_read(filepath), key="config")
 
+    def load_resource_file(
+        # special case where the environment variable keys in the 'config' value
+        # should not be replaced but preserved as is
+        self,
+        filepath: Path,
+        environment_variables: dict[str, str | None] | None = None,
+    ) -> list[dict[str, Any]]:
+        raw_str = self.safe_read(filepath)
+
+        original = load_yaml_inject_variables(raw_str, {}, validate=False, original_filepath=filepath)
+        replaced = load_yaml_inject_variables(raw_str, environment_variables or {}, original_filepath=filepath)
+
+        if isinstance(original, dict) and isinstance(replaced, dict):
+            if "config" in original:
+                replaced["config"] = original.get("config")
+                return [replaced]
+
+        for orig_item, repl_item in zip(original, replaced):
+            if "config" in orig_item:
+                repl_item["config"] = orig_item.get("config")  # type: ignore
+        return replaced  # type: ignore
```

Comment:
This mypy you cannot ignore, you have to respect the method signature.
```suggestion
        return replacedif isinstance(replaced, list) else [replaced]
```

Reply:
Ok, I'll accept that but it seems a bit strange since I know it has to be a list ;) 

```
if isinstance(original, dict) and isinstance(replaced, dict):
            if "config" in original:
                replaced["config"] = original.get("config")
                return [replaced]

        for orig_item, repl_item in zip(original, replaced):
            if "config" in orig_item:
                repl_item["config"] = orig_item.get("config")  # type: ignore
        return replaced  # type: ignore
```

Reply:
Sorry, I missed the other return. Got it.